src/gt4py/next/program_processors/runners/dace_fieldview/transformations/strides.py

```diff
@@ -24,6 +40,11 @@ def gt_change_transient_strides(
     transients in the optimal way.
     The function should run after all maps have been created.
 
+    After the strides have been adjusted the function will also propagate
+    the strides into nested SDFG. This propagation will happen with
+    `ignore_symbol_mapping` set to `True`, see `gt_propagate_strides_of()`
```

Comment:
I am of the opinion that `ignore_symbol_mapping` should be `False`. Actually, I would even prefer that this parameter was not available as configuration option. If the original SDFG was using symbol `A` for the stride of the inner array, the transformed SDFG should still use `A`, only the symbol mapping should change. Otherwise, if we replace symbols in nested SDFGs, we are doing two things in one transformation: stride propagation and symbol propagation.

Reply:
I have now thought a bit more about it and came to the conclusion that in certain situation `ignore_symbol_mapping=False` is actually wrong and will produce invalid results.
The reason is that using the symbol mapping actually affects the whole (nested) SDFG and not only the data descriptor.
The problem only occurs if different data descriptor use the same symbols as stride, which happens for example for temporaries, I mean for a long time they had the size `(horizontal_end, vertical_end)` which which would result in the strides `(horizontal_end, 1)`, which all transient shares.
This is the situation that `tests/next_tests/unit_tests/program_processor_tests/runners_tests/dace_tests/transformation_tests/test_strides.py::test_strides_propagation_shared_symbols_sdfg` demonstrates.
For some reasons the strides of only one descriptors (`a1`) are propagated, the other remains the same. If the symbol mapping would be used, then we would also change, as a side effect, the strides of `b2` (the inner view).

Earlier I have removed the symbol aliasing code, `2b03bb4`, in the `ignore_symbol_mapping=True` branch, exactly of that reason.
I also do not think that this is a big problem, as strides are not directly accessed, the shape is much more likely to be used, which does not change.

Reply:
You are right. Based on your explanation, I now agree on using `ignore_symbol_mapping=True` in its current meaning.

However, as a suggestion, we could change the meaning of `ignore_symbol_mapping`:
- `ignore_symbol_mapping=True` would behave as of current implementation: takes the outer strides and just apply it on the inner arrays.
- `ignore_symbol_mapping=False` would create new symbols in the nested SDFG (using the utility function to generate stride symbols based on the data name, and this should be enough to ensure that they are unique) and apply symbol mapping to the outer strides. The benefit of this is to avoid the risk of using a symbol that is already defined in the nested SDFG, with a different value than in parent SDFG.

Reply:
This is an issue indeed, I added a todo.